### PR TITLE
Use environment variables for AeDebugger mode

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
Use environment variables for AeDebugger mode

### CODE REVIEW

File: src/Microsoft.TestPlatform.Execution.Shared/ProcDumpExecutableHelper.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+
+using Microsoft.VisualStudio.TestPlatform.CoreUtilities;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.PlatformAbstractions;
+
```
Comment (Evangelink) [inline src/Microsoft.TestPlatform.Execution.Shared/ProcDumpExecutableHelper.cs:10]: Nit:

```suggestion
```

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcDumpDumper.cs [hunk 2]
```diff
@@ -86,7 +98,7 @@ public void AttachToTargetProcess(int processId, string outputDirectory, DumpTyp
             throw new InvalidOperationException("Procdump crash dump file must end with .dmp extension.");
         }
 
-        if (!TryGetProcDumpExecutable(processId, out var procDumpPath))
+        if (!new ProcDumpExecutableHelper(_processHelper, _fileHelper, _environment, _environmentVariableHelper).TryGetProcDumpExecutable(processId, out var procDumpPath))
```
Comment (Evangelink) [inline src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcDumpDumper.cs:101]: Might be better to new this up in ctor and reuse it.

File: src/vstest.console/Processors/AeDebuggerArgumentProcessor.cs [hunk 3]
```diff
@@ -123,29 +127,39 @@ private ArgumentProcessorResult InstallUnistallPostmortemDebugger(bool install)
...
-        // Validate ProcDumpToolDirectoryPath
-        if (!TryGetDirectoryInfo(_collectDumpParameters,
-            "ProcDumpToolDirectoryPath",
-            CommandLineResources.ProcDumpToolDirectoryPathArgumenNotFound,
-            CommandLineResources.InvalidProcDumpToolDirectoryPath,
-            out DirectoryInfo? procDumpToolDirectoryPath))
+        // Look for procdump
+        string? procDumpPath = null;
+        if (!TryGetDirectoryInfo(_collectDumpParameters, "ProcDumpToolDirectoryPath", out DirectoryInfo? procDumpToolDirectoryPath) &&
+            !new ProcDumpExecutableHelper(_processHelper, _fileHelper, _environment, _environmentVariableHelper).TryGetProcDumpExecutable(out procDumpPath)
```
Comment (Evangelink) [inline src/vstest.console/Processors/AeDebuggerArgumentProcessor.cs:133]: It's not changing much here but I would also new up the instance in ctor.

File: src/vstest.console/Resources/Resources.resx [hunk 4]
```diff
@@ -761,7 +761,7 @@
     <value>Postmortem debugger is not supported in the current OS.</value>
   </data>
   <data name="InvalidProcDumpToolDirectoryPath" xml:space="preserve">
-    <value>The directory specified is not valid: '{0}'</value>
+    <value>The directory specified for the procdump executable is not valid and the tool was not found inside environment variables(PROCDUMP_PATH, PATH)</value>
```
Comment (Evangelink) [inline src/vstest.console/Resources/Resources.resx:764]: Nit: 
```suggestion
    <value>The directory specified for the procdump executable is not valid and the tool was not found inside environment variables (PROCDUMP_PATH, PATH)</value>
```

**Review by Evangelink — COMMENTED** [review]
Will do a better pass next week but I am posting some small nitpicks already.

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcDumpDumper.cs (hunk 2 above)
Comment (MarcoRossignoli) [reply src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcDumpDumper.cs]: done

File: src/vstest.console/Processors/AeDebuggerArgumentProcessor.cs (hunk 3 above)
Comment (MarcoRossignoli) [reply src/vstest.console/Processors/AeDebuggerArgumentProcessor.cs]: done

**Review by Evangelink — APPROVED** [review]

**Pull request merged**